initial commit needs x-ray daemon containers updated

jasonrichardsmith opened the pull request:
*Issue #, if available:*
This provides a ui so users see App Mesh feeatures.

*Description of changes:*
Added new demo for meshdemo

This is waiting for this to be merged, so x-ray daemon images can be updated.
https://github.com/aws/aws-xray-daemon/pull/32

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

kiranmeduri commented on examples/apps/meshdemo/README.md:
```diff
@@ -0,0 +1,371 @@
+# Graphic AWS App Mesh Demo
```
is it possible to get a TOC here?

jasonrichardsmith replied:
added

kiranmeduri commented on examples/apps/meshdemo/README.md:
```diff
@@ -0,0 +1,371 @@
+
+All the code for the applications can be found in the cat, colors and frontend folders of this folder.
+
+### Running the services
```
Adding a number (chapter) will help here.

jasonrichardsmith replied:
I tried numbers they all looked weird to me.  Please review the current version and if you still want numbers I can add them.

kiranmeduri commented on examples/apps/meshdemo/README.md:
````diff
@@ -0,0 +1,371 @@
+
+To view the frontend we can port-forward the service using kubectl.
+```bash
+k port-forward -n meshdemo svc/frontend 8080:8000
````
you may need to add your aliases above :)

jasonrichardsmith replied:
I think I removed all aliases

kiranmeduri commented on examples/apps/meshdemo/kubernetes/app/kustomize/base/cats.yaml:
```diff
@@ -0,0 +1,134 @@
+      serviceAccountName: xray-access
+      containers:
+      - name: cats
+        image: 072792469044.dkr.ecr.eu-west-1.amazonaws.com/meshdemo/cats:1.0.0
```
will these be replaced by kustomize?

jasonrichardsmith replied:
I am open to hosting these images wherever.  Currently they are on my AWS internal account, but a quick change to the Makefile and send them anywhere, including docker hub.

kiranmeduri commented:
Can you please move this under walkthroughs/howto-k8s-meshdemo ?

lavignes commented on examples/apps/meshdemo/cats/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+FROM golang AS builder
+WORKDIR /go/src/github.com/jasonrichardsmith/meshdemo-cats
```
Should update the go module to be under aws/app-mesh-examples instead.

lavignes commented on examples/apps/meshdemo/frontend/Dockerfile:
```diff
@@ -0,0 +1,9 @@
+FROM node:9.11.1
```
Odd number node releases are unsupported after 6 months. Any reason why this isn't 12?

jasonrichardsmith replied:
I wrote the front end over 1.5 years ago, have not really changed it much since then.  I will update.

lavignes commented on examples/apps/meshdemo/kubernetes/app/kustomize/base/colors.yaml:
```diff
@@ -0,0 +1,255 @@
+          value: eu-west-1
+        - name: ENABLE_ENVOY_XRAY_TRACING
+          value: "1"
+        image: 840364872350.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.12.1.0-prod
```
We make an effort to not hardcode any envoy images I'm in this repo. Also v1.12.1.0 doesn't support IRSA.

jasonrichardsmith replied:
Agreed, I actually posted in the internal channel about this.  I had no insight into what the envoy container does and does not support specifically in regards to IRSA.  this is also why eksctl had to set the --appmesh flag.

jasonrichardsmith commented:
> Can you please move this under walkthroughs/howto-k8s-meshdemo ?

sure

jasonrichardsmith commented on examples/apps/meshdemo/Makefile:
```diff
@@ -0,0 +1,44 @@
+SHELL=/bin/bash -eo pipefail
+.DEFAULT_GOAL := build
+REGION=eu-west-1
```
I removed it from the deployments and put it into a configmap.

jasonrichardsmith commented on examples/apps/meshdemo/Makefile:
```diff
@@ -0,0 +1,44 @@
+SHELL=/bin/bash -eo pipefail
+.DEFAULT_GOAL := build
+REGION=eu-west-1
+ACCOUNT=072792469044
```
this is mone but I can move it anywhere

jasonrichardsmith commented on examples/apps/meshdemo/Makefile:
```diff
@@ -0,0 +1,44 @@
+ACCOUNT=072792469044
+REPOFAMILY=${ACCOUNT}.dkr.ecr.${REGION}.amazonaws.com/meshdemo
+VERSION=$(shell cat VERSION)
+HASH=$(shell git log --pretty=format:'%H' -n 1)
```
removed

jasonrichardsmith commented on examples/apps/meshdemo/README.md:
````diff
@@ -0,0 +1,371 @@
+
+You can run this to apply these changes:
+```bash
+k apply -k kubernetes/app/kustomize/cats-virtual-service
````
all should be fixed

jasonrichardsmith commented:
fixed a lot of issues, updated README can be found here:
https://github.com/jasonrichardsmith/aws-app-mesh-examples/tree/meshdemo/walkthroughs/howto-k8s-meshdemo

kiranmeduri commented:
Closing this due to staleness.
